src/lightning/pytorch/strategies/fsdp.py

```diff
@@ -396,3 +401,42 @@ def register_strategies(cls, strategy_registry: _StrategyRegistry) -> None:
             cpu_offload=True,
         )
         cls._registered_strategies.append("fsdp_cpu_offload")
+
+    def load_optimizer_state_dict(self, checkpoint: Mapping[str, Any]) -> None:
+        optimizer_states = checkpoint["optimizer_states"]
```

Comment:
Do checkpoints saved with FSDP before this PR have this key? Should we early exit if this key is not present? It might be worth doing also if we consider a future where https://github.com/Lightning-AI/lightning/issues/5339 is resolved.

Reply:
Updated